spec/lib/figaro_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+describe Figaro::Environment do
+  let(:config) do
+    {
+      'test_key' => 'value',
+      'production' => {
+        'test_key' => 'overridden value',
+      },
+    }
+  end
+
+  let(:config_environment) { 'test' }
+
+  before do
+    stub_const('ENV', {})
+  end
+
+  describe '#initialize' do
+    it 'warns and uses ENV when key is set in ENV and file config' do
+      stub_const('ENV', 'test_key' => 'overridden value')
```

Comment:
should we override it with a different value than `config` above so we actually confirm it's overriden?

Reply:
also we can jsut write to ENV here:

```suggestion
      ENV['test_key'] = 'aaa'
```